Yoast/wordpress-seo
# Show Gutenberg compatibility notifications if applicable

## Summary

This PR can be summarized in the following changelog entry:

* Adds notifications to the Notification Center in regards to Gutenberg compatibility. If Gutenberg is older than the minimum supported version by YoastSEO, a 'problem' notification is added. If Gutenberg is only slightly outdated, a 'normal' notification is added.

## Relevant technical choices:

*

## Test instructions

This PR can be tested by following these steps:

**Acceptance test only**
* Checkout this branch, make sure you have run `composer install`. 
* If you're running an environment with `YOAST_SEO_DEV_SERVER` set to true, ensure you also run `yarn`, `yarn upgrade yoast-components` and `yarn start`. Otherwise, you can skip this step.

**Testing steps**
* Ensure you have the latest version (https://downloads.wordpress.org/plugin/gutenberg.3.4.0.zip) of Gutenberg installed and activated.
* Go into SEO -> General and check that **no** notifications are being shown in regards to Gutenberg incompatibility.
* Uninstall Gutenberg and install an older version (https://downloads.wordpress.org/plugin/gutenberg.3.3.0.zip).
* Go back into SEO -> General and see that a notification has been added in regards to Gutenberg incompatibility under the Notifications header.
* Uninstall Gutenberg and install an _even older_ version (https://downloads.wordpress.org/plugin/gutenberg.2.2.0.zip).
* Go back into SEO -> General and see that a notification has been added in regards to Gutenberg incompatibility under the Problems header.

## Quality assurance


Fixes #10503
